internal/trie/node/decode.go

```diff
@@ -105,6 +105,17 @@ func decodeBranch(reader io.Reader, header byte) (branch *Branch, err error) {
 				ErrDecodeChildHash, i, err)
 		}
 
+		// Handle inlined leaf nodes.
+		if Type(hash[0]>>6) == LeafType && len(hash) != HashLength {
```

Comment:
nit it would look better to have this hash length as an inlined constant here, since it's not really used elsewhere :wink: 

```suggestion
		const hashLength = 32
		if Type(hash[0]>>6) == LeafType && len(hash) < hashLength {
```